### PR TITLE
Add installation of ALZ Powershell module for policy scripts

### PR DESCRIPTION
**Pull request by oZakari**
# Overview/Summary

Now that the Edit-LineEndings function has been moved to the ALZ PowerShell Module, I have adjusted the Invoke-PolicytoBicep scripts to prompt users to install the module.

Azure/Azure-Landing-Zones#2717

## This PR fixes/adds/changes/removes

1. Adds the installation of the ALZ-PowerShell-Module to the Invoke-PolicytoBicep.ps1 script.
1. Adds the installation of the ALZ-PowerShell-Module to the Invoke-PolicytoBicep-China.ps1 script and cleans up local installation of ALZ.Tools references

### Breaking Changes

1. None

## Testing Evidence

Performs test to ensure it prompts the user prior to installing and that it wouldn't prompt user again after the initial installation.

## As part of this Pull Request I have

- [x] Read the [Contribution Guide](https://github.com/Azure/ALZ-Bicep/wiki/Contributing) and ensured this PR is compliant with the guide
- [ ] Ensured the resource API versions in `.bicep` file/s I am adding/editing are using the latest API version possible
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/ALZ-Bicep/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/ALZ-Bicep/issues)
- [ ] *(ALZ Bicep Core Team Only)* Associated it with relevant [ADO Items](https://aka.ms/alz/bicep/backlog)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/ALZ-Bicep/tree/main)
- [x] Performed testing and provided evidence.
- [ ] Updated one or more of the following tests *(if required)*
  - [Unit](https://github.com/Azure/ALZ-Bicep/blob/main/.github/workflows/bicep-build-to-validate.yml)
  - [Linting](https://github.com/Azure/ALZ-Bicep/tree/main/.github/workflows)
  - [E2E (End-To-End)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/bicep-build-to-validate.yml)
  - [ValidateAzCloud (Base validation in Azure Cloud)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/base-unit-validate.yml)
  - [ValidateMcCloud (Base validation in Azure China Cloud)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/mc-base-unit-validate.yml)
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Module READMEs, Wiki Docs etc.)
- [ ] If relevant, created or updated Code Tours [here](https://github.com/Azure/ALZ-Bicep/blob/main/.vscode/tours)

### CODE REVIEW

**Comment by jtracey93** [comment]
@ozakari these scripts are invoked by https://github.com/Azure/ALZ-Bicep/blob/main/.github/workflows/update-policy.yml & https://github.com/Azure/ALZ-Bicep/blob/main/.github/workflows/update-policy-china.yml, do we need to install the ALZ module as part of these workflows? Or is that already covered?

**Comment by oZakari** [comment]
Thanks @jtracey93 for linking those, I've updated the MC workflow and added some logging as well.

File: .github/scripts/Invoke-PolicyToBicep-China.ps1 [hunk 1]
```diff
@@ -43,8 +41,13 @@ param (
 )
 
 # This script relies on a custom set of classes and functions
-# defined within the Alz.Tools PowerShell module.
-Import-Module $alzToolsPath -ErrorAction Stop
+# defined within the [ALZ-PowerShell-Module](https://github.com/Azure/Alz-powershell-module).
+if (-not (Get-Module -ListAvailable -Name ALZ)) {
+  # Module doesn't exist, so install it
+  Install-Module -Name ALZ -Scope CurrentUser -ErrorAction Stop
```
Comment (jtracey93) [inline .github/scripts/Invoke-PolicyToBicep-China.ps1:47]:
```suggestion
  Write-Information "====> ALZ module isn't already installed. Installing..." -InformationAction Continue
  Install-Module -Name ALZ -Force -Scope CurrentUser -ErrorAction Stop
  Write-Information "====> ALZ module now installed." -InformationAction Continue
```

File: .github/scripts/Invoke-PolicyToBicep.ps1 [hunk 2]
```diff
@@ -38,6 +38,15 @@ param (
   $assignmentsTxtFileName = "_policyAssignmentsBicepInput.txt"
 )
 
+# This script relies on a custom set of classes and functions
+# defined within the [ALZ-PowerShell-Module](https://github.com/Azure/Alz-powershell-module).
+if (-not (Get-Module -ListAvailable -Name ALZ)) {
+  # Module doesn't exist, so install it
+  Install-Module -Name ALZ -Scope CurrentUser -ErrorAction Stop
```
Comment (jtracey93) [inline .github/scripts/Invoke-PolicyToBicep.ps1:45]:
```suggestion
  Write-Information "====> ALZ module isn't already installed. Installing..." -InformationAction Continue
  Install-Module -Name ALZ -Force -Scope CurrentUser -ErrorAction Stop
  Write-Information "====> ALZ module now installed." -InformationAction Continue
```

**Review by jtracey93 — CHANGES_REQUESTED** [review]
couple of tweaks

**Pull request merged**